node/src/ActiveSpeakerObserver.ts

```diff
@@ -18,9 +19,18 @@ export interface ActiveSpeakerObserverActivity {
 	producer: Producer;
 }
 
+type ObserverEvents = RtpObserverEvents & {
+	dominantspeaker: [{ producer: Producer }];
+}
+
+type Events = {
+	routerclose: [];
+	dominantspeaker: [{ producer: Producer }];
+}
```

Comment:
Some concerns here that affects all files and not just this one:

1. Can we please define first class events and then observer events?.
2. `routerclose` is not just an event in `ActiveSpeakerObserver` class. It's an event that all `RtpObserver` inherited classes.
3. The name of `RtpObserver` class in unfortunate since it also contains an `observer`. So events of the `observer` of `RtpObserver` should be named `RtpObserverObserverEvents`.
4. And events of the `RtpObserver` parent class should be named `RtpObserverEvents`.
5. Some classes (such as `Transport` and `RtpObserver` export they `XxxxxxEvents` and `XxxxxObserverEvents`. However, in child classes this PR defines `type Events` and `type ObserverEvents`. For consistency we should export these as well and prefix their name with the class name.
6. Not sure why ESLint doesn't complain but we mandate opening brackets to be in the next line.

So IMHO this should be as follows:

```suggestion
type ActiveSpeakerEvents = RtpObserverEvents &
{
	dominantspeaker: [{ producer: Producer }];
}

type ActiveSpeakerObserverEvents = RtpObserverObserverEvents &
{
	dominantspeaker: [{ producer: Producer }];
}
```